src/main/java/org/mifos/connector/common/zeebe/ZeebeProcessStarter.java

```diff
@@ -0,0 +1,121 @@
+package org.mifos.connector.common.zeebe;
+
+import io.camunda.zeebe.client.ZeebeClient;
+import io.camunda.zeebe.client.api.response.ProcessInstanceEvent;
+import org.mifos.connector.common.util.MpesaUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+@Component
+public class ZeebeProcessStarter {
+    private static Logger logger = LoggerFactory.getLogger(ZeebeProcessStarter.class);
+
+    @Autowired
+    private ZeebeClient zeebeClient;
+
+    @Value("${transaction-id-length}")
+    private int transactionIdLength;
+
+
+    public String startZeebeWorkflow(String workflowId, String request, Map<String, Object> extraVariables) {
+        String transactionId = generateTransactionId();
+
+        Map<String, Object> variables = new HashMap<>();
+        variables.put(ZeebeVariables.TRANSACTION_ID, transactionId);
+        variables.put(ZeebeVariables.CHANNEL_REQUEST, request);
+        variables.put(ZeebeVariables.ORIGIN_DATE, Instant.now().toEpochMilli());
+        if (extraVariables != null) {
+            variables.putAll(extraVariables);
+        }
+
+        logger.info("starting workflow HERE:");
+        // TODO if successful transfer response arrives in X timeout return it otherwise do callback
+        ProcessInstanceEvent instance = zeebeClient.newCreateInstanceCommand()
+                .bpmnProcessId(workflowId)
+                .latestVersion()
+                .variables(variables)
+                .send()
+                .join();
+
+        logger.info("zeebee workflow instance from process {} started with transactionId {}, instance key: {}", workflowId, transactionId, instance.getProcessInstanceKey());
+        return transactionId;
+    }
+
+    public String startMpesaZeebeWorkflow(String workflowId, String request, Map<String, Object> extraVariables) {
```

Comment:
I think this has been deprecated, please check

Reply:
This function is used to start a workflow